api/user.py

```diff
@@ -106,14 +106,58 @@ def register():
     response.set_cookie('session_id', session_id, httponly=True) # FIXME: Get rid of secure=True as development server does not support HTTPS
     return response, 200
 
+@users_bp.route("/change-password", methods=['PUT'])
+@authenticate()
+def change_password():
+    data = request.json
+    old_password: str = data.get('old_password')
+    password: str = data.get('password')
+    user: User = request.user
+
+    if (len(password) < 10 or len(password) > 64):
+        return jsonify({"error": "New password must be between 10 to 64 characters"}), 400
+    if (len(old_password) < 10 or len(old_password) > 64):
+        return jsonify({"error": "Old password should be between 10 to 64 characters"}), 400
+
+    if not verify_password(user.password, old_password, user.salt):
+        return jsonify({"error": "old password incorrect"}), 400
+
+    salt = generate_salt()
+    hashed_new_password = hash_password(password, salt)
+
+    try:
+        user = db_client.session.query(User).filter_by(email=user.email).first()
```

Comment:
The user from `request.user` should already be the user queried from db, so you can directly change things on the user without requerying for it